src/main/java/io/github/debuggyteam/architecture_extensions/api/BlockType.java

```diff
@@ -60,12 +61,14 @@ public String toString() {
 		return name().toLowerCase(Locale.ROOT);
 	}
 
-	public TypedGroupedBlock register(BlockGroup.GroupedBlock groupedBlock, Set<ItemGroup> itemGroups) {
+	public TypedGroupedBlock register(BlockGroup group, BlockGroup.GroupedBlock groupedBlock, ArchExIntegrationContextImpl.BlockCreationCallback onBlockCreated) {
 		var id = ArchitectureExtensions.id(groupedBlock.id().getPath() + "_" + this);
-		var block = Registry.register(Registries.BLOCK, id, creator.apply(groupedBlock.baseBlock(), QuiltBlockSettings.copyOf(groupedBlock.baseBlock()).mapColorProvider(state -> groupedBlock.mapColor()).strength(strength)));
+		var baseBlock = groupedBlock.baseBlock().get();
+		var block = Registry.register(Registries.BLOCK, id, creator.apply(baseBlock, QuiltBlockSettings.copyOf(baseBlock).mapColorProvider(state -> groupedBlock.mapColor()).strength(strength)));
 
-		var item = Registry.register(Registries.ITEM, id, new BlockItem(block, new QuiltItemSettings()));
-		for (ItemGroup itemGroup : itemGroups) { ItemGroupUtil.pull(itemGroup, this, groupedBlock.baseBlock(), item); }
+		Registry.register(Registries.ITEM, id, new BlockItem(block, new QuiltItemSettings()));
+		
+		onBlockCreated.onBlockCreated(group, this, baseBlock, block);
```

Comment:
Random comment: `onBlockCreated.onBlockCreated` feels a bit odd; might be better as `blockCreationCallback` like the interface name? Not really major.